src/mem/physical.cc

```diff
@@ -463,21 +463,21 @@ PhysicalMemory::unserializeStore(CheckpointIn &cp)
               range_size, range.size());
 
     uint64_t curr_size = 0;
-    long* temp_page = new long[chunk_size];
-    long* pmem_current;
+    int64_t* temp_page = new int64_t[chunk_size];
```

Comment:
@giactra do you think it makes sense to use Addr here?

Reply:
Honestly, I don't know why we are not allocating with bytes here (using uint8_t)

Reply:
Yes! That makes more sense. To answer you question, it's because this code was committed in 2006, lol